packages/astro/src/core/routing/params.ts

```diff
@@ -27,12 +27,12 @@ export function getParams(array: string[]) {
  * values and create a stringified key for the route
  * that can be used to match request routes
  */
-export function stringifyParams(params: Params, routeComponent: string) {
+export function stringifyParams(params: GetStaticPathsItem['params'], routeComponent: string) {
 	// validate parameter values then stringify each value
 	const validatedParams = Object.entries(params).reduce((acc, next) => {
 		validateGetStaticPathsParameter(next, routeComponent);
 		const [key, value] = next;
-		acc[key] = typeof value === 'undefined' ? undefined : `${value}`;
+		acc[key] = value?.toString();
```

Comment:
This is more concise and works for `undefined`, numbers, and strings, but does require support for the `?.` operator. Since this support was added in Node.js 14.0.0, this should be OK.